Tasks/JavaToolInstaller/javatoolinstaller.ts

```diff
@@ -0,0 +1,101 @@
+import taskLib = require('vsts-task-lib/task');
+import toolLib = require('vsts-task-tool-lib/tool');
+import path = require('path');
+
+import { AzureStorageArtifactDownloader } from "./AzureStorageArtifacts/AzureStorageArtifactDownloader";
+import { JavaFilesExtractor } from "./FileExtractor/JavaFilesExtractor";
+taskLib.setResourcePath(path.join(__dirname, 'task.json'));
+
+async function run() {
+    try {
+        let versionSpec = taskLib.getInput('versionSpec', true);
+        await getJava(versionSpec);
+    }
+    catch (error) {
+        console.error('ERR:' + error.message);
+    } 
+}
+
+async function getJava(versionSpec: string) {
+    toolLib.debug('Trying to get tool from local cache');
+    let localVersions: string[] = toolLib.findLocalToolVersions('Java');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+    let fromAzure: boolean = ('AzureStorage' == taskLib.getInput('jdkSource', true));
+    let fromLocalDirectory: boolean = ('LocalDirectory' == taskLib.getInput('jdkSource', true));
```

Comment:
Any reason to have two variables for this?  If this is a picklist, and it's one or the other, any reason why we should test the two variables?  Ex.
`if (fromAzure) {
} else {
// from local
}`